### PR TITLE
update all the things

### PR DESCRIPTION
**Pull request by onemen**
first step for #450 

this PR contains :
* Remix v1.17.1 v2 dev server + HMR
* All other dependencies updated

on my Windows 11 all test are passing except `Call Kent recording flow`

### CODE REVIEW

File: app/components/sections/hero-section.tsx [hunk 1]
```diff
@@ -93,6 +93,7 @@ function HeroSection({
           {imageProps ? (
             <motion.img
               {...imageProps}
+              suppressHydrationWarning={true}
```
Comment (kentcdodds) [inline app/components/sections/hero-section.tsx]: Why is this necessary?

File: package.json [hunk 2]
```diff
@@ -65,65 +65,72 @@
...
-    "metascraper": "^5.34.4",
-    "metascraper-description": "^5.34.4",
-    "metascraper-image": "^5.34.4",
-    "metascraper-title": "^5.34.4",
+    "metascraper": "^5.34.7",
+    "metascraper-description": "^5.34.7",
+    "metascraper-image": "^5.34.7",
+    "metascraper-title": "^5.34.7",
     "morgan": "^1.10.0",
+    "node-fetch": "^3.3.1",
```
Comment (kentcdodds) [inline package.json:115]: Why was this dependency added?

File: package.json [hunk 3]
```diff
@@ -65,65 +65,72 @@
...
     "on-finished": "^2.4.1",
     "p-limit": "^4.0.0",
     "p-props": "^5.0.0",
     "p-queue": "^7.3.4",
     "patch-package": "^7.0.0",
     "path-to-regexp": "^6.2.1",
-    "pm2": "^5.3.0",
```
Comment (kentcdodds) [inline package.json:117]: I'm thrilled to remove this.

**Review by kentcdodds — CHANGES_REQUESTED** [review]
This is wonderful! Thank you for working on this! I have a few comments and questions.

File: app/components/sections/hero-section.tsx (hunk 1 above)
Comment (onemen) [reply app/components/sections/hero-section.tsx]: i get this on localhost
```
react-dom.development.js:86 Warning: Prop `alt` did not match. Server: "Illustration of Kody the Koala skiing surrounded by green leaves, a battery, two skies, a one-wheel, a solar panel, and a recycle logo." Client: "Illustration of Kody the Koala standing on a snowboard surrounded by green leaves, a battery, two skies, a one-wheel, a solar panel, and a recycle logo."
    at img
    at MotionComponent (http://localhost:3000/build/_shared/chunk-BKXH5CFN.js:285:40)
    at div
    at div
    at header
    at Grid2 (http://localhost:3000/build/_shared/chunk-ZH4MEJF3.js:40:3)
    at HeroSection (http://localhost:3000/build/_shared/chunk-BKXH5CFN.js:7553:3)
    at div
    at IndexRoute (http://localhost:3000/build/routes/index-EVZDMZW4.js:683:16)
    at RemixRoute (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:5762:3)
    at RenderedRoute (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:3483:5)
    at RenderErrorBoundary (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:4041:9)
    at Outlet (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:3825:26)
    at body
    at html
    at App (http://localhost:3000/build/_shared/chunk-FSYMWBNO.js:7734:16)
    at ThemeProvider (http://localhost:3000/build/_shared/chunk-PS2XUJXE.js:54:3)
    at TeamProvider (http://localhost:3000/build/_shared/chunk-FSYMWBNO.js:6138:3)
    at AppWithProviders (http://localhost:3000/build/_shared/chunk-FSYMWBNO.js:7855:16)
    at RemixRoute (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:5762:3)
    at RenderedRoute (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:3483:5)
    at RenderErrorBoundary (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:4041:9)
    at DataRoutes (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:3753:5)
    at Router (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:3832:15)
    at RouterProvider (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:3702:5)
    at RemixErrorBoundary (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:5352:5)
    at RemixBrowser (http://localhost:3000/build/_shared/chunk-HIPMZLZ5.js:7277:40)
```

File: package.json (hunk 2 above)
Comment (onemen) [reply package.json:115]: `generate-build-info.js`
https://github.com/kentcdodds/kentcdodds.com/blob/e755dfbef26b9beb21616133fc4c86fb613bbd51/other/generate-build-info.js#L1-L5

File: app/components/sections/hero-section.tsx (hunk 1 above)
Comment (kentcdodds) [reply app/components/sections/hero-section.tsx]: That warning is an indication of a real problem. That alt text should be exactly the same in the client and server. If it's not, the user will see a flash of the wrong image. I can look at this later if you don't get to it first, but we need to determine which image/alt text to use in the loader so it remains consistent.

Comment (onemen) [reply app/components/sections/hero-section.tsx]: I don't see it on the real site, only on my local dev build. 
Maybe it is only window issue. Maybe it is related to \\ instead of / somewher?

**Comment by onemen** [comment]
currently there is an issue with `refresh-on-content-changes.js` it is running but it does not refresh the cash for the file that was change

**Comment by onemen** [comment]
@kentcdodds ,

After [85ec064](https://github.com/kentcdodds/kentcdodds.com/pull/451/commits/85ec064a30bd9144e227d063e47afc7994cc4f81)
the cache refresh, and remix HMR triggered, however the page content does not update without hitting F5

```shell
[dev:remix  ] GET localhost:3000/blog/stop-being-a-junior 200  - 78.436 ms
[dev:content] changed C:\projects\kcd-apps\kentcdodds.com\content\blog\stop-being-a-junior.mdx
[dev:remix  ] POST localhost:3000/action/refresh-cache 200  - 8.616 ms
[dev:content] Content change request finished. {
[dev:content]   response: {
[dev:content]     message: 'Refreshing cache for content paths',
[dev:content]     contentPaths: [ 'blog/stop-being-a-junior.mdx' ]
[dev:content]   }
[dev:content] }
[dev:remix  ] Updated the cache value for blog:dir-list. Getting a fresh value for this took 174ms. Caching for 1209600000ms + 2592000000ms stale in SQLite cache.
[dev:remix  ] Updated the cache value for blog:mdx-list-items. Getting a fresh value for this took 370ms. Caching for 1209600000ms + 2592000000ms stale in SQLite cache.
[dev:remix  ] 💿 File created: app\refresh.ignored.js
[dev:remix  ] Updated the cache value for blog:stop-being-a-junior:downloaded. Getting a fresh value for this took 391ms. Caching for 1209600000ms + 2592000000ms stale in SQLite cache.
[dev:remix  ] 💿 Rebuilding...
[dev:remix  ] Updated the cache value for blog:stop-being-a-junior:compiled. Getting a fresh value for this took 816ms. Caching for 1209600000ms + 2592000000ms stale in SQLite cache.
[dev:remix  ] Updated the cache value for mdx-page:blog:stop-being-a-junior:compiled. Getting a fresh value for this took 1215ms. Caching for 1209600000ms + 2592000000ms stale in SQLite cache.
[dev:remix  ] 💿 Rebuilt in 1.7s
[dev:remix  ] Waiting for app server (6a7e96b7)
[dev:remix  ] App server took 803ms
[dev:remix  ] > HMR
```

**Comment by kentcdodds** [comment]
I wonder if the loader re-runs before the cache has finished updating 🤔

**Comment by onemen** [comment]
> I wonder if the loader re-runs before the cache has finished updating ??

i've added log to `blog.tsx` and `blog.$slug.tsx` loader functions.
both loader did not run after `dev:content` trigger cache update

**Comment by kentcdodds** [comment]
Oh, that's probably because the way HMR works. Before it just did a full-page refresh, now it's only reloading the affected loaders which is none of them I think. We'll have to re-think how to make this work 🤔

**Comment by onemen** [comment]
maybe we can use the same method you are using in `kcdshop` to reload after README file changes?

**Comment by kentcdodds** [comment]
We could do that. Or we could just trigger a save on the `blog.$slug.tsx` file. I'm pretty sure you can just save it without changing its contents. I've done that before. I think I did that in the remix repo...

Yep, something like this should do it: https://github.com/remix-run/remix/blob/a6173a1b1b6574d9d0fa2dce79f5f85cbbbec76c/rollup.utils.js#L64-L84

**Comment by kentcdodds** [comment]
For us we wouldn't be touching the entry.server file or anything (it appears that doesn't work anymore). Instead we'll do that to the `blog.$slug.tsx` file. Or whatever relevant file for the content we're editing.

**Comment by onemen** [comment]
strangely, when i start the server and navigate to a blog page that i've change, it show the original page before the change. it look like it dose not 'remember; the cache update,

**Comment by onemen** [comment]
we need to trigger `HMR + HDR` not just `HMR`

**Comment by kentcdodds** [comment]
> strangely, when i start the server and navigate to a blog page that i've change, it show the original page before the change. it look like it dose not 'remember; the cache update,

Could that be the browser cache? I think we've got cache headers enabled during development.

> we need to trigger `HMR + HDR` not just `HMR`

I'm not sure whether this is possible. We may be forced to trigger a full-page refresh like we do with kcdshop

**Comment by onemen** [comment]
@kentcdodds ,

check https://github.com/kentcdodds/kentcdodds.com/pull/451/commits/a9bc540bc6bd11d9f893cae0eb0f0aa174b0309a

this is based on the code from `kcdshop` and it work.

maybe there is a better way, is it possible to add the content watcher in the root and call fetcher to `action/refresh-cache` and when it complete to trigger the `window.location.reload()` or `navigate('.', {replace: true})`

File: app/components/sections/hero-section.tsx (hunk 1 above)
Comment (kentcdodds) [reply app/components/sections/hero-section.tsx]: Let's remove it for now and I'll look into it later.

File: app/root.tsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import {cssBundleHref} from '@remix-run/css-bundle/dist/index'
```
Comment (kentcdodds) [inline app/root.tsx]: Hmmm... is this really necessary? Can we not do this:

```suggestion
import {cssBundleHref} from '@remix-run/css-bundle'
```

That seems to work in the Epic Stack.

File: app/root.tsx [hunk 5]
```diff
@@ -537,6 +539,11 @@ function App() {
           }}
         />
         {ENV.NODE_ENV === 'development' ? <LiveReload nonce={nonce} /> : null}
+        <script
+          nonce={nonce}
+          suppressHydrationWarning
+          dangerouslySetInnerHTML={{__html: getWebsocketJS()}}
+        />
```
Comment (kentcdodds) [inline app/root.tsx]: This should be rendered in development only.

File: server/dev-server.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  } else {
+    const command =
+      'tsx watch --clear-screen=false --ignore "app/**" --ignore "build/**" --ignore "node_modules/**" --inspect ./index.js'
+    execa(command, {
+      stdio: ['ignore', 'inherit', 'inherit'],
+      shell: true,
+      env: {
+        ...process.env,
+        FORCE_COLOR: true,
+        MOCKS: true,
```
Comment (kentcdodds) [inline server/dev-server.js]: Let's make it possible to override these values if necessary

```suggestion
        FORCE_COLOR: true,
        MOCKS: true,
        ...process.env,
```

**Review by kentcdodds — COMMENTED** [review]
This is looking really good. Just a few things.

File: app/components/sections/hero-section.tsx (hunk 1 above)
Comment (onemen) [reply app/components/sections/hero-section.tsx]: https://github.com/kentcdodds/kentcdodds.com/blob/94ae5849251a3a48e38d8253854730b712cf615b/app/routes/index.tsx#L87-L89

`getRandomFlyingKody(data.kodyTeam)` return random image.

I think this is the cause to this issue. can i move the line to the loader ?

Comment (kentcdodds) [reply app/components/sections/hero-section.tsx]: Ah, yeah, let's move that to the loader. Not sure how that got there! Thank you for finding it :)

**Comment by onemen** [comment]
https://github.com/kentcdodds/kentcdodds.com/pull/451/commits/0695a561b2d2281b274a4eb762d03798812164af
typescript didn't like it when i tried to move `getRandomFlyingKody` to the loader 
i did not want to rename the images file to images.server

i will handle the rest of your comments tomorrow

**Comment by onemen** [comment]
@kentcdodds 

this is mostly done.

```shell
[@octokit/plugin-throttling] `onAbuseLimit()` is deprecated and will be removed in a future release of `@octokit/plugin-throttling`, please use the `onSecondaryRateLimit` handler instead
```
 I am not sure i can just replace `onAbuseLimit` with `onSecondaryRateLimit`.
 can you take a look at it, maybe handle it in a separate PR

**Review by kentcdodds — APPROVED** [review]
This looks awesome. Thanks so much!

**Comment by kentcdodds** [comment]
Let's address the octokit thing later.
